src/models/Users.ts

```diff
@@ -0,0 +1,38 @@
+export enum Role {
+    Patient = "patient",
+    Doctor = "doctor",
+    Admin = "admin"
+}
+
+class User {
+    id: number | undefined;
+
+    email: string | undefined;
+
+    firstName: string | undefined;
+    lastName: string | undefined;
+}
+
+class Doctor extends User { }
+
+class Admin extends User { }
+
+class Address {
+    id: number | undefined;
+    city: string | undefined;
+    zipCode: string | undefined;
+    street: string | undefined;
+    houseNumber: string | undefined;
+    localNumber: string | undefined;
+}
```

Comment:
to samo, typy wrzuciłbym w /components/types.tsx (albo oddzielne directory na nie)
Poza tym tam już jest typ "Address"